src/core/annotations/qgsannotationpictureitem.h

```diff
@@ -0,0 +1,257 @@
+
+    /**
+     * Sets the \a unit to use for fixed picture sizes, when the sizeMode() is Qgis::AnnotationPictureSizeMode::FixedSize.
+     *
+     * \see fixedSizeUnit()
+     * \see setFixedSize()
+     */
+    void setFixedSizeUnit( Qgis::RenderUnit unit );
+
+    /**
+     * Returns TRUE if the aspect ratio of the picture will be retained.
+     *
+     * \see setLockAspectRatio()
+     */
+    bool lockAspectRatio() const;
+
+    /**
+     * Sets whether the aspect ratio of the picture will be retained.
+     *
+     * \see lockAspectRatio()
+     */
+    void setLockAspectRatio( bool locked );
+
+    /**
+     * Returns TRUE if the item's background should be rendered.
+     *
+     * \see setBackgroundEnabled()
+     * \see backgroundSymbol()
+     */
+    bool backgroundEnabled() const { return mDrawBackground; }
+
+    /**
+     * Sets whether the item's background should be rendered.
+     *
+     * \see backgroundEnabled()
+     * \see setBackgroundSymbol()
+     */
+    void setBackgroundEnabled( bool enabled ) { mDrawBackground = enabled; }
+
+    /**
+     * Returns the symbol used to render the item's background.
+     *
+     * \see backgroundEnabled()
+     * \see setBackgroundSymbol()
+     */
+    const QgsFillSymbol *backgroundSymbol() const;
+
+    /**
+     * Sets the \a symbol used to render the item's background.
+     *
+     * The item takes ownership of the symbol.
+     *
+     * \see backgroundSymbol()
+     * \see setBackgroundEnabled()
+     */
+    void setBackgroundSymbol( QgsFillSymbol *symbol SIP_TRANSFER );
+
+    /**
+     * Returns TRUE if the item's frame should be rendered.
+     *
+     * \see setFrameEnabled()
+     * \see frameSymbol()
+     */
+    bool frameEnabled() const { return mDrawBorder; }
+
+    /**
+     * Sets whether the item's frame should be rendered.
+     *
+     * \see frameEnabled()
+     * \see setFrameSymbol()
+     */
+    void setFrameEnabled( bool enabled ) { mDrawBorder = enabled; }
+
+    /**
+     * Returns the symbol used to render the item's frame.
+     *
+     * \see frameEnabled()
+     * \see setFrameSymbol()
+     */
+    const QgsFillSymbol *frameSymbol() const;
+
+    /**
+     * Sets the \a symbol used to render the item's frame.
+     *
+     * The item takes ownership of the symbol.
+     *
+     * \see frameSymbol()
+     * \see setBackgroundEnabled()
+     */
+    void setFrameSymbol( QgsFillSymbol *symbol SIP_TRANSFER );
+
+  private:
+
+    QString mPath;
+    Qgis::PictureFormat mFormat = Qgis::PictureFormat::Unknown;
+    Qgis::AnnotationPictureSizeMode mSizeMode = Qgis::AnnotationPictureSizeMode::SpatialBounds;
+    QgsRectangle mBounds;
+
+    QSizeF mFixedSize;
+    Qgis::RenderUnit mFixedSizeUnit = Qgis::RenderUnit::Millimeters;
+
+    bool mLockAspectRatio = true;
+    bool mDrawBackground = false;
+    std::unique_ptr< QgsFillSymbol > mBackgroundSymbol;
+    bool mDrawBorder = false;
+    std::unique_ptr< QgsFillSymbol > mBorderSymbol;
```

Comment:
Would be nice to align the member name to their paired functions.

```suggestion
    bool mDrawFrame = false;
    std::unique_ptr< QgsFillSymbol > mFrameSymbol;
```

Reply:
Done